fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanStatus.java

```diff
@@ -80,6 +82,42 @@ public static LoanStatus fromInt(final Integer statusValue) {
         return enumeration;
     }
 
+    public static LoanStatus fromString(final String statusString) {
+
+        LoanStatus enumeration = LoanStatus.INVALID;
+
+        if (StringUtils.isEmpty(statusString)) {
+            return enumeration;
+        }
+
+        if (statusString.equalsIgnoreCase(LoanStatus.SUBMITTED_AND_PENDING_APPROVAL.toString())) {
```

Comment:
@thesmallstar I'm probably missing something, but isn't this just `LoanStatus.valueOf(statusString)`? If you're doing this only because of `equalsIgnoreCase()` then wouldn't a `LoanStatus.valueOf(statusString.toUpperCase())` suffice and have the same effect? If you want an invalid status String to return LoanStatus.INVALID, you may have to `catch (IllegalArgumentException e)` .. (I think; not 100% sure, do check). Actually that is debatable.. wouldn't you rather that an invalid status filter passed through the REST API from a client lead to an error, than be silently ignored?

Reply:
For the first part, yes, they would have the same effect. I copied this implementation from some other enum and hence the code looks like this and not how you suggested. Do you see any advantage of one over the other?

2nd part answer in a comment below.

Reply:
Just another point: given this validation seems to be a common pattern in the enums, I've added a method in the DataValidatorBuilder (isOneOfEnumValues()) to check if the value provided is valid for a specific enum (and report back with correct PlatformApiDataException if it isn't).

So one way to handle these is in the validation phase check the values using that, after which it should be safe to use valueOf() without getting an exception. 

So as far as I can see there are three ways to do the same thing:

- custom valueOf method with error handling to avoid an exception (as currently in many places in the code), and then check for invalid status afterwards - i.e. the current logic in this PR

- call valueOf and catch the exception if the value is invalid, and rethrow as PlatformApiDataException

- call DataValidatorBuilder.isOneOfEnumValues() to validate and then valueOf to convert

Not sure there's a significant advantage of one way over another...